newton/_src/utils/schema_resolver.py

```diff
@@ -0,0 +1,479 @@
+            "limit_angular_kd": [AttrSpec("physxLimit:angular:damping")],
+            "limit_rotX_kd": [AttrSpec("physxLimit:rotX:damping")],
+            "limit_rotY_kd": [AttrSpec("physxLimit:rotY:damping")],
+            "limit_rotZ_kd": [AttrSpec("physxLimit:rotZ:damping")],
+            "angular_position": [AttrSpec("state:angular:physics:position")],
+            "linear_position": [AttrSpec("state:linear:physics:position")],
+            "rotX_position": [AttrSpec("state:rotX:physics:position")],
+            "rotY_position": [AttrSpec("state:rotY:physics:position")],
+            "rotZ_position": [AttrSpec("state:rotZ:physics:position")],
+            "angular_velocity": [AttrSpec("state:angular:physics:velocity")],
+            "linear_velocity": [AttrSpec("state:linear:physics:velocity")],
+            "rotX_velocity": [AttrSpec("state:rotX:physics:velocity")],
+            "rotY_velocity": [AttrSpec("state:rotY:physics:velocity")],
+            "rotZ_velocity": [AttrSpec("state:rotZ:physics:velocity")],
+        },
+        "shape": {
+            # Mesh hull vertex limit
+            "mesh_hull_vertex_limit": [AttrSpec("physxConvexHullCollision:hullVertexLimit")],
+            # Collision contact offset
+            "collision_contact_offset": [AttrSpec("physxCollision:contactOffset")],
+        },
+        "material": {
+            "stiffness": [AttrSpec("physxMaterial:compliantContactStiffness")],
+            "damping": [AttrSpec("physxMaterial:compliantContactDamping")],
+        },
+        "body": {
+            # Rigid body damping
+            "rigid_body_damping": [AttrSpec("physxRigidBody:linearDamping"), AttrSpec("physxRigidBody:angularDamping")],
+        },
+    }
+
+
+def _solref_to_stiffness(solref):
+    """Convert MuJoCo solref (timeconst, dampratio) to internal stiffness.
+
+    k = 1 / (timeconst^2)
+    """
+    try:
+        timeconst = float(solref[0])
+        dampratio = float(solref[1]) if len(solref) > 1 else None
+    except Exception:
+        return None
+    # Direct mode: both negative → interpret as (damping, stiffness)
+    if timeconst is not None and timeconst < 0.0 and dampratio < 0.0:
+        return -timeconst
+    if not (timeconst and timeconst > 0.0):
+        return None
+    return 1.0 / (timeconst * timeconst)
+
+
+def _solref_to_damping(solref):
+    """Convert MuJoCo solref (timeconst, dampratio) to internal damping.
+
+    b = 2 * dampratio / timeconst
+    """
+    try:
+        timeconst = float(solref[0])
+        dampratio = float(solref[1])
+    except Exception:
+        return None
+    # Direct mode: both negative → interpret as (damping, stiffness)
+    if timeconst < 0.0 and dampratio < 0.0:
+        return -dampratio
+    if not (dampratio and dampratio > 0.0):
+        return None
+    return (2.0 * dampratio) / timeconst
+
+
+class MjcPlugin(EngineSchemaPlugin):
+    name: ClassVar[str] = "mjc"
+
+    mapping: ClassVar[dict[str, dict[str, list[AttrSpec]]]] = {
```

Comment:
We should probably ask someone from GDM to review this `mapping`, the `_solref_to_*` functions, and their applicability to each of the properties below